### DISCUSSION TITLE
Umzug von Raspbian 4 auf NUC

### DISCUSSION THREAD

**Discussion by woupi** [camueller/SmartApplianceEnabler#186] [Installation und Betrieb]
Hallo,
erstmal nochmals vielen Dank an Axel, für deine ganze Arbeit, da kann man sich, meine ich, nicht oft genug für bedanken.
Nachdem nun das ganze etwas sortierter hier in github läuft möchte ich auch gleich nochmal um Rat bitten.

Ich habe letztes Jahr den Raspbian 4 installiert und hatte leider viele Probleme mit den Stromzählern. Die ersten Modbus von Orno habe ich und auch Axel nie ans laufen bekommen und sie sind darum auch wieder zurück gegangen. Dann habe ich die SDM120Modbus gekauft. Aber auch die zeigten sich zickig und so habe ich diese leider auch nicht ans rennen bekommen. Nur die unermüdliche Arbeit von Axel ist es zu verdanken, dass sie nun laufen.

Jetzt möchte ich, wie im Betreff angedeutet, das ganze auf einen NUC umziehen.

Dank einiger Tipps aus dem Photovoltaikforum, habe ich dort nun Proxmox am laufen. Außerdem läuft auch SAE bereits in einem LXC Container. Ebenso iobroker in einem anderen.

Nun muss ich so langsam auch mal den Modbus USB Adapter vom Raspbian in den NUC stecken und SAE konfigurieren.

Leider habe ich damals bei dem ganzen rumprobieren, nicht alles dokumentiert.

Welche Dateien müsste ich sichern?

Von der mbusd-ttyUSB0.conf, habe ich mir damals ein Screenshot gemacht. 
Ich gehe davon aus, das diese erst erstellt wird, wenn ein USB Adapter angeschlossen wird, da zur Zeit auf der SAE Installation vom NUC es noch keine Datei mit dem Namen gibt.

Danke schonmal für die Hilfe.

**Comment by camueller** [comment]
Du musst `mbusd` installieren wie in der Doku beschrieben: https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Modbus_DE.md

**Comment by woupi** [comment]
Danke, stimmt, da muss ich die ja erstellen, sorry für die blöde frage, hätte ich auch selber drauf kommen können.
Andere Dateien müssen nicht gesichert werden, entnehme ich deiner Antwort, oder?
mbusd wollte ich erst installieren, wenn ich den Stick um gesteckt habe. 
Werde ich dann aber heute  Abend mal machen.

**Comment by woupi** [comment]
Leider startet mbusd nicht und mir sagen die Fehler leider auch nichts.
Für Tipps wäre ich Dankbar.

sae@SAE:/etc/mbusd$ sudo systemctl status mbusd@ttyUSB0.service
x mbusd@ttyUSB0.service - Modbus TCP to Modbus RTU (RS-232/485) gateway.
     Loaded: loaded (/lib/systemd/system/mbusd@.service; disabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Sat 2022-01-29 16:16:48 UTC; 21s ago
    Process: 442 ExecStart=/usr/bin/mbusd -d -v9 -c /etc/mbusd/mbusd-ttyUSB0.conf -p /dev/ttyUSB0 (code=exited, status=2)
   Main PID: 442 (code=exited, status=2)
        CPU: 899us

Jan 29 16:16:47 SAE systemd[1]: mbusd@ttyUSB0.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Jan 29 16:16:47 SAE systemd[1]: mbusd@ttyUSB0.service: Failed with result 'exit-code'.
Jan 29 16:16:48 SAE systemd[1]: mbusd@ttyUSB0.service: Scheduled restart job, restart counter is at 5.
Jan 29 16:16:48 SAE systemd[1]: Stopped Modbus TCP to Modbus RTU (RS-232/485) gateway..
Jan 29 16:16:48 SAE systemd[1]: mbusd@ttyUSB0.service: Start request repeated too quickly.
Jan 29 16:16:48 SAE systemd[1]: mbusd@ttyUSB0.service: Failed with result 'exit-code'.
Jan 29 16:16:48 SAE systemd[1]: Failed to start Modbus TCP to Modbus RTU (RS-232/485) gateway..


Im log steht:
29 Jan 2022 16:16:43 conn_init() failed, exiting...
29 Jan 2022 16:16:45 mbusd-0.3.1 started...
29 Jan 2022 16:16:45 conn_init(): can't open tty device /dev/ttyUSB0 (No such file or directory)

**Comment by woupi** [comment]
Hat sich erledigt, ich habe einfach mal neu gestartet und nun lässt sich mbusd auch starten.

Jetzt geht es weiter

Zu früh gefreut, nach dem ich 
sudo systemctl enable mbusd@ttyUSB0.service
ausgeführt habe und zum testen einen reboot gemacht habe, kommt nun folgender Fehler:

sae@SAE:~$ sudo systemctl status mbusd@ttyUSB0.service
* mbusd@ttyUSB0.service - Modbus TCP to Modbus RTU (RS-232/485) gateway.
     Loaded: loaded (/lib/systemd/system/mbusd@.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Sat 2022-01-29 16:52:18 UTC; 962ms ago
    Process: 226 ExecStart=/usr/bin/mbusd -d -v2 -L - -c /etc/mbusd/mbusd-ttyUSB0.conf -p /dev/ttyUSB0 (code=exited, status=2)
   Main PID: 226 (code=exited, status=2)
        CPU: 837us

Jan 29 16:52:18 SAE systemd[1]: mbusd@ttyUSB0.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Jan 29 16:52:18 SAE systemd[1]: mbusd@ttyUSB0.service: Failed with result 'exit-code'.

**Comment by Trunks1982** [comment]
Versuchst du nen usb in den lxc durchzureichen?
Ist nicht ganz so einfach.Deswegen würde ich dir auch empfehlen iob in einer vm zu installieren und alles darüber zum sae mittels simple api zu schicken.

**Comment by woupi** [comment]
Ja, genau, das versuche ich. Danke für den Hinweis. Dan  werde ich wohl wieder von vorne anfangen müssen.

**Comment by Trunks1982** [comment]
Geht ja fix.Nimm am besten unbuntu ohne gui

**Comment by Trunks1982** [comment]
Ach übrigens benutze ich nen orno fehlerfrei über iob🤣

**Reply by camueller** [reply]
Eigentlich sollte es egal sein, welchen Modbus-Zähler man verwendet. Wenn ich mich richtig erinnere, wurde bei Orno immer irgendein Windows-Konfigprogramm mitgeliefert anstatt einer ordentlichen Dokumentation der Register (wie das z.B. bei b+g e-Tech bzw. eastron der Fall ist)

**Reply by Trunks1982** [reply]
Eigentlich schon.Aufjedenfall wenn einer die Register haben will, gibt einfach bescheid.Oder er wird hier mit in die Doku aufgenommen.

**Reply by woupi** [reply]
Mit aufnehmen in der Doku wäre bestimmt für einige 👍,  zumal es im Moment mal wieder die sdm120modbus bei b&g nicht gibt.

**Comment by woupi** [comment]
Guten Morgen,  danke nochmal für den Hinweis,  leider stehe ich damit vor den nächsten Problemen. 
Ich habe keine Ahnung was simpel api ist und eine kurze Recherche in Google hat mir leider auch nicht die Erleuchtung gebracht. 
Bleibt sae im lxc Container oder wo muss ich sae nun installieren?
Ich würde mich wie immer über jede Hilfe freuen.

**Reply by Trunks1982** [reply]
Der sae bleibt im lxc.

**Comment by woupi** [comment]
Ich habe heute morgen eine VM mit Ubuntu server aufgesetzt und darauf nun iobroker installiert.
Bei google habe ich nach "iobroker simple api zu lxc durchreichen" und nach "proxmox lxc usb durchreichen simple api" gesucht, beides hat mich leider nicht weiter gebracht.

Wärst Du so lieb mir zu erklären, was ich da genau machen muss?

**Reply by Trunks1982** [reply]
Ja gerne.Aber sag mir jetzt erstmal was du vorhast?
Zählerdaten usw von io nach sae?

**Comment by woupi** [comment]
Ja genau,.

**Reply by Trunks1982** [reply]
Wenn du magst können wir mal über anydesk ne session machen.Geht so schneller und du verstehst es einfacher.

**Comment by woupi** [comment]
Sehr gerne, bin nur leider gerade mit dem Hund raus, hättest du in einer Stunde  noch Zeit?

**Reply by Trunks1982** [reply]
Ja sollte klappen.

**Comment by woupi** [comment]
👍danke

**Comment by woupi** [comment]
Hallo,
ich bin jetzt wieder an dem Rechner und habe auch noch schnell anydesk installiert, ich bin also bereit, wenn Du Zeit hast, könnten wir los legen.

**Reply by Trunks1982** [reply]
wäre auch bereit.bräuchte jetzt nur deine nr
kannste dann auch direkt hier wieder löschen

**Comment by woupi** [comment]


**Reply by Trunks1982** [reply]
hab dir im pv forum was geschrieben.

**Comment by woupi** [comment]
Dank Trunks1982, läuft nun der Nuc mit Proxmox, einer VM mit IoBroker, einem lxc Container mit Sae und die Verbindung von dem Modbus Stromzähler zum SHM. Danke dafür.
Falls einer das liest, ich habe versucht alles zu dokumentieren und werde das auch nochmal sauber schreiben und würde das natürlich bei Interesse auch zur Verfügung stellen.

**Comment by ImWe4sel** [comment]
Hallo Woupi. gibt es bereits einen Draft deiner Doku? Ich scheitere momentan an der Installation im lxc container und bin über jede HIlfe dankbar

**Comment by woupi** [comment]
Hallo ImWe4sel,

ja, ich habe da was zusammen geschrieben, veröffentlichen würde ich das so allerdings noch nicht.
Allerdings geht es bei dem was ich ich geschrieben habe auch nicht um die Installation von SAE, sondern um die Verbindung von Modbus Zähler über iobroker zu SAE.

Nichtsdestotrotz habe ich natürlich auf dem NUC Proxmox am laufen und darauf auch SAE im lxc installiert.

Eventuell kann ich Dir also ja trotzdem Helfen, wenn Du mir sagst, wo Du genau stehst, bzw wo es hängt.

**Reply by ImWe4sel** [reply]
Hallo Woupi, 
danke für das Angebot, aber SAE läuft jetzt endlich auch bei mir ;) Ich hoffe auch dass das so bleibt, ansonsten nehm ich dein Angebot gerne an!

**Comment by woupi** [comment]
OK, dann habe ich dich wohl falsch verstanden, wo liegt den dann genau dein Problem, oder hast Du das Problem nur in der zwischen Zeit selbst gelöst?
